tests/rptest/tests/data_transforms_test.py

```diff
@@ -1164,34 +1165,62 @@ def test_manager_metrics_values(self):
         self.redpanda.set_cluster_config(
             {'data_transforms_logging_flush_interval_ms': 60 * 60 * 1000})
 
+        cfg = self._admin.get_cluster_config()
+
+        buf_capacity_B = cfg.get(
+            'data_transforms_logging_buffer_capacity_bytes', 0)
+        assert buf_capacity_B > 0, "Expected non-zero log buffer capacity"
+
+        max_log_line_B = cfg.get('data_transforms_logging_line_max_bytes', 0)
+        assert max_log_line_B > 0, "Expected non-zero max log line"
+
+        self.logger.debug(
+            f"Buffer capacity: {buf_capacity_B}B, Max log line: {max_log_line_B}B"
+        )
+
         it, ot = self.setup_identity_xform(
             self.topics[0],
             self.topics[1],
             name=f"logger-xform",
         )
 
         N_PER_TP = 32
+        k = self.random_ascii_string(128, 128)
+        min_log_usage_B = N_PER_TP * max_log_line_B
         for _ in range(0, N_PER_TP):
             self._rpk.produce(
                 it.name,
-                self.random_ascii_string(128, 128),
-                self.random_ascii_string(128, 128),
+                k,
+                self.random_ascii_string(max_log_line_B, max_log_line_B),
             )
 
-        self._rpk.consume(ot.name, n=1, offset=N_PER_TP - 1)
+        def get_buffer_usage() -> list[float]:
+            all_nodes_usage: list[float] = []
+            for node in self.redpanda.nodes:
+                samples = self.unpack_samples(
+                    self.get_metrics_from_node(
+                        node,
+                        ["log_manager_buffer_usage"],
+                        endpoint=MetricsEndpoint.METRICS,
+                    ))
 
-        all_nodes_usage = []
-        for node in self.redpanda.nodes:
-            samples = self.unpack_samples(
-                self.get_metrics_from_node(
-                    node,
-                    ["log_manager_buffer_usage"],
-                    endpoint=MetricsEndpoint.METRICS,
-                ))
+                for s in samples['log_manager_buffer_usage']:
+                    all_nodes_usage.append(s['value'])
+            self.logger.debug(f"MAX: {max(all_nodes_usage)}")
+            return all_nodes_usage
 
-            for s in samples['log_manager_buffer_usage']:
-                all_nodes_usage.append(s['value'])
+        expected_usage = ((min_log_usage_B / buf_capacity_B)  #
+                          * 0.95)  # slop
 
-        assert any(
-            bu > 0.0 for bu in all_nodes_usage
-        ), f"Expected some non-zero buffer usage, got {all_nodes_usage}"
+        self.logger.debug(
+            f"Should eventually reach {100 * expected_usage}% usage on some node"
+        )
+
+        wait_until(lambda: any(bu > expected_usage
+                               for bu in get_buffer_usage()),
```

Comment:
These are transient and shard local. If leadership changes then this test will fail.

TBH I would think it's fine to just remove the test entirely, or soften the assertion so that if we send double the expected data we never go over the limit (which is really more what we care for)